### ISSUE TITLE
(Makefile) remove-spaces command does not work on GNU system

### ISSUE THREAD

**Issue by damsien** [kubernetes-sigs/kubebuilder#4327]
### What broke? What's expected?

This issue is related to the `Makefile` of Kubebuilder and **not** to the generated `Makefile` used for the operator.

The command `make remove-spaces` executed on a GNU system raises an error.
https://github.com/kubernetes-sigs/kubebuilder/blob/e9beffb5926f8fcf1abf97ada7ae3a3c675d0591/Makefile#L71-L74

That is due to a difference between the `sed` of the GNU system and the OS X system (this command works well on MacOS). It is well explained [here](https://unix.stackexchange.com/a/131940).

My proposal to solve this is to add a check of the system.
```
.PHONY: remove-spaces
remove-spaces:
	@echo "Removing trailing spaces"
        @SED_CMD="sed -i ''" && [ "$$(uname)" != "Darwin" ] && SED_CMD="sed -i"; \
	find . -type f -name "*.md" -exec $$SED_CMD 's/[[:space:]]*$$//' {} + || true
```

### KubeBuilder (CLI) Version

4.3.1

**Comment by camilamacedo86** [comment]
that is working in the CI (which is ubuntu)
but I am fine with any change that works for mac and linux
feel free to open a pr with your suggestion

**Comment by damsien** [comment]
From what I saw in the `.github/workflows`, there is no call to this `make remove-spaces`. This command is only called once by the `make generate` command.
https://github.com/kubernetes-sigs/kubebuilder/blob/20a99551e2e7c7681949e0f08bbc3594b6249ef4/Makefile#L66-L69

Correct me if I am wrong but the `make generate` command is never executed during the CI process.

PS: the "Trailing spaces" CI job only check for trailing spaces without trying to remove them (`make test-spaces`).
https://github.com/kubernetes-sigs/kubebuilder/blob/20a99551e2e7c7681949e0f08bbc3594b6249ef4/.github/workflows/spaces.yml#L11-L21

**Comment by damsien** [comment]
Should we keep this issue open? I wanted to show you the error. In the following screenshot, the first command is executed without the "fix" I made. And the second command is executed with the "fix".

![image](https://github.com/user-attachments/assets/fa6dc39b-62ba-4649-a466-cf5b3fc10691)

The fix I made is wrong. I can try to find a friend that have a mac to actually test the command or wait for someone that want to take the issue.

**Comment by camilamacedo86** [comment]
HI @damsien 

The way that it is today I know that work on Mac
So, could we not:

If is mac then do as it is today
else
do the way that worked for you in linux

WDYT?
Do you want to try it?
I can test locally the next one.

**Comment by damsien** [comment]
Yes we can do this, I'm not going to create the PR today but I'll work on it

**Comment by camilamacedo86** [comment]
solved
